Extension/src/Debugger/attachQuickPick.ts

```diff
@@ -30,44 +30,42 @@ export interface AttachItem extends vscode.QuickPickItem {
     id?: string;
 }
 
-export function showQuickPick(getAttachItems: () => Promise<AttachItem[]>): Promise<string | undefined> {
-    return getAttachItems().then(processEntries =>
-        new Promise<string>((resolve, reject) => {
-            const quickPick: vscode.QuickPick<AttachItem> = vscode.window.createQuickPick<AttachItem>();
-            quickPick.title = localize("attach.to.process", "Attach to process");
-            quickPick.canSelectMany = false;
-            quickPick.matchOnDescription = true;
-            quickPick.matchOnDetail = true;
-            quickPick.placeholder = localize("select.process.attach", "Select the process to attach to");
-            quickPick.items = processEntries;
-            quickPick.buttons = [new RefreshButton()];
-
-            const disposables: vscode.Disposable[] = [];
-
-            quickPick.onDidTriggerButton(button => {
-                getAttachItems().then(processEntries => quickPick.items = processEntries);
-            }, undefined, disposables);
-
-            quickPick.onDidAccept(() => {
-                if (quickPick.selectedItems.length !== 1) {
-                    reject(new Error(localize("process.not.selected", "Process not selected.")));
-                }
+export async function showQuickPick(getAttachItems: () => Promise<AttachItem[]>): Promise<string | undefined> {
+    const processEntries: AttachItem[] = await getAttachItems();
+    return new Promise<string | undefined>((resolve, reject) => {
+        const quickPick: vscode.QuickPick<AttachItem> = vscode.window.createQuickPick<AttachItem>();
+        quickPick.title = localize("attach.to.process", "Attach to process");
+        quickPick.canSelectMany = false;
+        quickPick.matchOnDescription = true;
+        quickPick.matchOnDetail = true;
+        quickPick.placeholder = localize("select.process.attach", "Select the process to attach to");
+        quickPick.items = processEntries;
+        quickPick.buttons = [new RefreshButton()];
+
+        const disposables: vscode.Disposable[] = [];
+
+        quickPick.onDidTriggerButton(async () => { quickPick.items = await getAttachItems(); }, undefined, disposables);
```

Comment:
I think the first call to `getAttachItems` was unnecessary. `const processEntries: AttachItem[] = await getAttachItems();`
I removed the first call.

Reply:
> Not your change, but do you know if this code is actually necessary? quickPick.items was already populated by a call to getAttachItems(), and it don't see any code here that would cause it to return something different the second time.

Yes. This was to handle refreshing using the refresh button and not having to close out of the quickPickMenu and restart a attach launch. It will be different based on the output of `wmic` or `ps`.

See https://github.com/microsoft/vscode-cpptools/pull/2895/files

The first call is important for the window to be initially populated.